controllers/horizon_controller.go

```diff
@@ -340,6 +344,23 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 		return ctrl.Result{}, err
 	}
 
+	if instance.Spec.SharedMemcached == "" {
+		memcachedExists, err := r.getMemcachedCR(ctx, fmt.Sprintf("%s-memcached", instance.Name))
+		if err != nil {
+			r.Log.Info("Failed retrieving Memcached CR list")
+			return ctrl.Result{}, err
+		}
```

Comment:
If you know that SharedMemcached is not provided in the CR then you don't need to try to look up the Memcached either and you can move ahead and create your own.

Reply:
I believe the latest update resolves this concern